### PR TITLE
Add K3s component documentation page

### PR DESCRIPTION
**Pull request by diconico07**

### CODE REVIEW

File: asciidoc/components/k3s.adoc [hunk 1]
```diff
@@ -9,4 +9,34 @@ ifdef::env-github[]
...
 endif::[]
 
-https://docs.k3s.io
+https://k3s.io/[K3s] is a highly available, certified Kubernetes distribution designed for production workloads in unattended, resource-constrained, remote locations or inside IoT appliances.
+
+It is packaged as a single and small binary so installations and updates are fast and easy.
+
+== How does SUSE Edge use K3s
+
+The use of K3s as the SUSE Edge stack Kubernetes distribution is only recommended when etcd as a backend is not a possibility. If you can use etcd as a backend, use link:./rke2.adoc[RKE2].
```
Comment (agracey) [inline asciidoc/components/k3s.adoc:18]: While I agree with this statement, we may want a paragraph before it that's a little gentler?

File: asciidoc/components/k3s.adoc [hunk 2]
```diff
@@ -9,4 +9,34 @@ ifdef::env-github[]
...
+
+== Best practices
+
+=== Installation
+The recommended way of installing K3s as part of the SUSE Edge stack is by using Edge Image Builder (EIB), see link:./edge-image-builder.adoc[its documentation] for more details on how to configure it to deploy K3s.
+
+It automatically supports HA setup, as well as Elemental setup.
+
+=== Fleet for GitOps workflow
```
Comment (agracey) [inline asciidoc/components/k3s.adoc:27]: Good call out! This really helps connect the whole story

**Review by agracey — APPROVED** [review]
Thank you. 

I added a few comments but we can handle those in the edit phase

File: asciidoc/components/k3s.adoc [hunk 3]
```diff
@@ -9,4 +9,34 @@ ifdef::env-github[]
...
+=== Storage management
+
+K3s comes with local-path storage pre-configured, this is suitable for single-node clusters.
+For clusters spanning over multiple nodes, it is recommended to use link:./longhorn.adoc[Longhorn]
+
+=== Load Balancing and HA
+
+If you installed K3s using EIB this part is already covered by the EIB documentation in the HA section.
+
+Otherwise, you need to install and configure MetalLB as per our link:./metallb[MetalLB documentation].
```
[the remark this answers is not in the document]
Comment (agracey) [reply asciidoc/components/k3s.adoc:40]: Ok, just merged that branch so if you pull main, the right pages should exist to link against!

File: asciidoc/components/k3s.adoc (hunk 1 above)
Comment (diconico07) [reply asciidoc/components/k3s.adoc:18]: Added a small paragraph before this one, hopefully gentler :innocent:

**Review by agracey — APPROVED** [review]
LGTM

**Pull request merged**